reason-parser/src/reason_pprint_ast.ml

```diff
@@ -125,6 +125,7 @@ and ruleCategory =
      that it's easier just to always wrap them in parens.  *)
   | PotentiallyLowPrecedence of layoutNode
```

Comment:
@let-def Not urgent, but in case you stumble upon this: this `ruleCategory` type is the closest thing to an intermediate representation that we already have to work with. You could decide to evolve it to be more sophisticated, or start fresh eventually.